### PR TITLE
Feature/user skills

### PR DESCRIPTION
**Pull request by kopkop123**

### CODE REVIEW

File: frontend/package.json [hunk 1]
```diff
@@ -29,6 +29,7 @@
     "react-redux": "7.2.4",
     "react-router-dom": "5.2.0",
     "react-scripts": "4.0.3",
+    "react-select": "^4.3.1",
```
Comment (aleksandr-kuharenko) [inline frontend/package.json]:
```suggestion
    "react-select": "4.3.1",
```

File: frontend/package.json [hunk 2]
```diff
@@ -40,6 +41,7 @@
     "@types/react-dom": "16.9.14",
     "@types/react-redux": "7.1.18",
     "@types/react-router-dom": "5.1.8",
+    "@types/react-select": "^4.0.17",
```
Comment (aleksandr-kuharenko) [inline frontend/package.json]:
```suggestion
    "@types/react-select": "4.0.17",
```

File: frontend/src/components/profile/components/profile-edit/profile-edit.tsx [hunk 3]
```diff
@@ -71,6 +109,32 @@ const ProfileEdit: React.FC = () => {
     }
   };
 
+  const handleInputChange = (inputValue: any): void => {
```
Comment (aleksandr-kuharenko) [inline frontend/src/components/profile/components/profile-edit/profile-edit.tsx]: Specify type

File: frontend/src/components/profile/components/profile-edit/profile-edit.tsx [hunk 4]
```diff
@@ -71,6 +109,32 @@ const ProfileEdit: React.FC = () => {
...
+    const lastSkill = inputValue[inputValue.length - 1];
+    if(lastSkill.__isNew__) {
+      skillApi.createSkill(lastSkill.value).then((response: any) => {
```
Comment (aleksandr-kuharenko) [inline frontend/src/components/profile/components/profile-edit/profile-edit.tsx]: Specify type

File: shared/src/common/interfaces/skill/skill.interface.ts [hunk 5]
```diff
@@ -0,0 +1,10 @@
+interface ISkill {
+  id?: string;
+  name?: string;
+  skills?: string;
+  __isNew__?: boolean;
```
Comment (aleksandr-kuharenko) [inline shared/src/common/interfaces/skill/skill.interface.ts:5]: where you receive this field?

Comment (kopkop123) [reply shared/src/common/interfaces/skill/skill.interface.ts:5]: This field receives from a React Select plugin, and needs to define a new skill

File: frontend/src/store/auth/actions.ts [hunk 6]
```diff
@@ -21,7 +21,7 @@ const login = createAsyncThunk(
 const register = createAsyncThunk(
   ActionType.SetUser,
   async (
-    registerPayload: Omit<IUser, 'id' | 'avatar'>,
+    registerPayload: Omit<IUser, 'id' | 'avatar' | 'title' | 'skills'>,
```
Comment (ursarik) [inline frontend/src/store/auth/actions.ts]: maybe we can use here [IRegister](https://github.com/BinaryStudioAcademy/bsa-2021-infostack/blob/feature/user-skills/shared/src/common/interfaces/auth/register.interface.ts) interface?

File: frontend/src/store/auth/actions.ts [hunk 7]
```diff
@@ -8,7 +8,7 @@ import { LocalStorageVariable } from 'common/enums/enums';
 const login = createAsyncThunk(
   ActionType.SetUser,
   async (
-    loginPayload: Omit<IUser, 'id' | 'fullName' | 'avatar'>,
+    loginPayload: Omit<IUser, 'id' | 'fullName' | 'avatar' | 'title' | 'skills'>,
```
Comment (ursarik) [inline frontend/src/store/auth/actions.ts]: we have [ILogin](https://github.com/BinaryStudioAcademy/bsa-2021-infostack/blob/feature/user-skills/shared/src/common/interfaces/auth/login.interface.ts) interface

File: frontend/src/services/skill-api/skill-api.service.ts [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+  }
+
+  public async createSkill(name: string): Promise<ISkill[]> {
+    const skillResponse: ISkill[] = await this.http.load(this.BASE, {
+      method: HttpMethod.POST,
+      payload: JSON.stringify({ name }),
+      contentType: ContentType.JSON,
+    });
+
+    return skillResponse;
```
Comment (ursarik) [inline frontend/src/services/skill-api/skill-api.service.ts]:
```suggestion
    return this.http.load(this.BASE, {
      method: HttpMethod.POST,
      payload: JSON.stringify({ name }),
      contentType: ContentType.JSON,
    });
```

File: frontend/src/services/auth-api/auth-api.service.ts [hunk 9]
```diff
@@ -8,7 +8,7 @@ const http = new Http();
 
 class AuthApi {
   public async loginUser(
-    loginPayload: Omit<IUser, 'id' | 'fullName' | 'avatar'>,
+    loginPayload: Omit<IUser, 'id' | 'fullName' | 'avatar' | 'title' | 'skills'>,
```
Comment (ursarik) [inline frontend/src/services/auth-api/auth-api.service.ts]: use [ILogin](https://github.com/BinaryStudioAcademy/bsa-2021-infostack/blob/feature/user-skills/shared/src/common/interfaces/auth/login.interface.ts) interface

File: frontend/src/services/auth-api/auth-api.service.ts [hunk 10]
```diff
@@ -20,7 +20,7 @@ class AuthApi {
   }
 
   public async registerUser(
-    registerPayload: Omit<IUser, 'id' | 'avatar'>,
+    registerPayload: Omit<IUser, 'id' | 'avatar' | 'title' | 'skills'>,
```
Comment (ursarik) [inline frontend/src/services/auth-api/auth-api.service.ts]: use [IRegister](https://github.com/BinaryStudioAcademy/bsa-2021-infostack/blob/feature/user-skills/shared/src/common/interfaces/auth/register.interface.ts) interface

**Review by aleksandr-kuharenko — APPROVED** [review]

File: frontend/src/services/auth-api/auth-api.service.ts [hunk 11]
```diff
@@ -8,7 +9,7 @@ const http = new Http();
 
 class AuthApi {
   public async loginUser(
-    loginPayload: Omit<IUser, 'id' | 'fullName' | 'avatar'>,
+    loginPayload: Omit<ILogin, 'id' | 'fullName' | 'avatar' | 'title' | 'skills'>,
```
Comment (ursarik) [inline frontend/src/services/auth-api/auth-api.service.ts]:
```suggestion
    loginPayload: ILogin
```

File: frontend/src/services/auth-api/auth-api.service.ts [hunk 12]
```diff
@@ -1,5 +1,6 @@
 import { IRefreshToken } from 'common/interfaces/auth';
-import { IUserWithTokens, IUser } from 'common/interfaces/user';
+import { IUserWithTokens } from 'common/interfaces/user';
+import { ILogin, IRegister } from '../../../../shared/build';
```
Comment (ursarik) [inline frontend/src/services/auth-api/auth-api.service.ts]: fix import

File: frontend/src/services/auth-api/auth-api.service.ts [hunk 13]
```diff
@@ -20,7 +21,7 @@ class AuthApi {
   }
 
   public async registerUser(
-    registerPayload: Omit<IUser, 'id' | 'avatar'>,
+    registerPayload: Omit<IRegister, 'id' | 'avatar' | 'title' | 'skills'>,
```
Comment (ursarik) [inline frontend/src/services/auth-api/auth-api.service.ts]:
```suggestion
    registerPayload: IRegister
```

File: frontend/src/store/auth/actions.ts [hunk 14]
```diff
@@ -2,13 +2,13 @@ import { createAsyncThunk } from '@reduxjs/toolkit';
 import { actions } from './slice';
 import { ActionType } from './common';
 import { AuthApi, UserApi } from 'services';
-import { IUser } from 'common/interfaces/user';
+import { ILogin, IRegister } from '../../../../shared/build';
```
Comment (ursarik) [inline frontend/src/store/auth/actions.ts]: fix import

File: frontend/src/store/auth/actions.ts [hunk 15]
```diff
@@ -2,13 +2,13 @@ import { createAsyncThunk } from '@reduxjs/toolkit';
...
 import { LocalStorageVariable } from 'common/enums/enums';
 
 const login = createAsyncThunk(
   ActionType.SetUser,
   async (
-    loginPayload: Omit<IUser, 'id' | 'fullName' | 'avatar'>,
+    loginPayload: Omit<ILogin, 'id' | 'fullName' | 'avatar' | 'title' | 'skills'>,
```
Comment (ursarik) [inline frontend/src/store/auth/actions.ts]:
```suggestion
    loginPayload: ILogin
```

File: frontend/src/store/auth/actions.ts [hunk 16]
```diff
@@ -21,7 +21,7 @@ const login = createAsyncThunk(
 const register = createAsyncThunk(
   ActionType.SetUser,
   async (
-    registerPayload: Omit<IUser, 'id' | 'avatar'>,
+    registerPayload: Omit<IRegister, 'id' | 'avatar' | 'title' | 'skills'>,
```
Comment (ursarik) [inline frontend/src/store/auth/actions.ts]:
```suggestion
    registerPayload: IRegister
```

**Review by ursarik — APPROVED** [review]

**Pull request merged**